src/common/utils/companion.ts

```diff
@@ -0,0 +1,73 @@
+import { PROTOCOL_TOKEN_ADDRESS } from '@common/mocks/tokens';
+import { FullPosition, Permission, Position } from '@types';
+
+export const getCompanionNeedsWithdraw = (position?: Nullable<FullPosition | Position>) => {
+  if (!position) {
+    return false;
+  }
+
+  const toHasYield = position.to.underlyingTokens.length;
+  const toIsProtocol = position.to.address === PROTOCOL_TOKEN_ADDRESS;
+
+  if (toHasYield || toIsProtocol) {
+    return true;
+  }
+
+  return false;
+};
+
+export const getCompanionNeedsIncreaseOrReduce = (position?: Nullable<FullPosition | Position>) => {
+  if (!position) {
+    return false;
+  }
+
+  const fromHasYield = position.from.underlyingTokens.length;
+  const fromIsProtocol = position.from.address === PROTOCOL_TOKEN_ADDRESS;
+
+  if (fromHasYield || fromIsProtocol) {
+    return true;
+  }
+
+  return false;
+};
+
+export const getCompanionNeedsTerminate = (position?: Nullable<FullPosition | Position>) => {
+  if (!position) {
+    return false;
+  }
+
+  const needsIncreaseOrReduce = getCompanionNeedsIncreaseOrReduce(position);
+  const needsWithdraw = getCompanionNeedsWithdraw(position);
+
+  if (needsIncreaseOrReduce || needsWithdraw) {
+    return true;
+  }
+
+  return false;
+};
```

Comment:
I don't remember our architecture, but, didn't we have a special terminate permission? I'm not sure if this logic makes sense, but it's mostly because I don't remember.

Reply:
Yup it does, this function checks if we need to have it